### PR TITLE
refactor: separate demo app code from app components

### PR DESCRIPTION
**Pull request by targos**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,9 +2,18 @@
...
+      "types": "./lib-esm/app/index.d.ts",
+      "require": "./lib/app/index.js",
+      "default": "./lib-esm/app/index.js"
+    },
+    "./components": {
+      "types": "./lib-esm/components/index.d.ts",
+      "require": "./lib/components/index.js",
+      "default": "./lib-esm/components/index.js"
+    }
+  },
```
Comment (targos) [inline package.json:16]: @stropitek We need to find better names for top-level directories/exports.
The existing separation is confusing:
- `./app` contains components
- `./components` contains things that are not components (hooks for example)

Comment (lpatiny) [reply package.json:16]: could `./app` be just `./demo` ?

Comment (targos) [reply package.json:16]: No, IMO the name doesn't make sense, and I just created the demo directory in this PR to separate the public app API from the demo app built from this repo.

**Comment by targos** [comment]
@zakodium-oss/collaborators @zakodium-oss/external-collaborators 

Note the relatively big changes I made in this PR:

- Move test/demo data to a top-level folder (test-data).
- Move demo-specific code to src/demo.
- Separate app data code from app compenents (create a new app-data folder).
- Reorganize src/components with logical subfolders.
- Export three top-level paths from the package:
  - `./ui` used for lower-level UI components (maps to src/components).
  - `./app` used for higher-level analysis components (such as panels, maps to src/app).
  - `./app-data` used for data loading and processing (maps to src/app-data).

**Pull request merged**